### PR TITLE
Integrate Trust Wallet [500 USDC Reward!]

### PR DESCRIPTION
**Pull request by isreallee82**

### CODE REVIEW

**Comment by MaxMillington** [comment]
@isreallee82 is this building for you locally? It fails the build step on Vercel
Screenshot 2024-02-08 at 9 26 44 AM

**Comment by isreallee82** [comment]
sorry for late response didnt get a notification, its not failing locally @MaxMillington

**Comment by isreallee82** [comment]
I will rerun checks @MaxMillington

**Comment by MaxMillington** [comment]
@isreallee82 Can you rebase and fix the conflicts and repush? Perhaps that will fix the broken build

**Comment by isreallee82** [comment]
alright I will do that @MaxMillington

**Comment by isreallee82** [comment]
a min @MaxMillington

**Comment by isreallee82** [comment]
@MaxMillington, I suspect that the recent update to cosmokit-core may be causing some issues with the type definitions. I'll take some time to review the changes made in their documentation before the weekend to troubleshoot and resolve any potential conflicts.

**Review by y3fers0n — APPROVED** [review]

**Comment by isreallee82** [comment]
sure I will @MaxMillington

**Comment by MaxMillington** [comment]
@isreallee82 just checking in on this

**Comment by isreallee82** [comment]
@MaxMillington still checking should have a positive feedback this weekend

File: packages/web/config/generate-cosmos-kit-wallet-list.ts
Comment (JoseRFelix) [inline packages/web/config/generate-cosmos-kit-wallet-list.ts:84]: Any reason why we are adding `any` here? I believe every registry object should match the Wallet type

**Review by JoseRFelix — CHANGES_REQUESTED** [review]

File: packages/web/config/generate-cosmos-kit-wallet-list.ts
Comment (MaxMillington) [reply packages/web/config/generate-cosmos-kit-wallet-list.ts:84]: cc @isreallee82

**Comment by DavideSegullo** [comment]
It seems like you can't switch between two different mnemonics, trust wallet doesn't provide an event for that

**Comment by CryptoAssassin1** [comment]
Getting this error on BLD, IST token deposit/withdraw.
image

**Comment by isreallee82** [comment]
@CryptoAssassin1 According to [Trust Doc](https://developer.trustwallet.com/developer/develop-for-trust/browser-extension), Trust Wallet Browser Extension currently supports only Ethereum & EVM chains, and support for Cosmos is still in progress.

File: packages/web/config/generate-cosmos-kit-wallet-list.ts
Comment (isreallee82) [reply packages/web/config/generate-cosmos-kit-wallet-list.ts:84]: @JoseRFelix, it appears that the mismatched types are causing the build to fail. Although upgrading wallet-core resolves this issue, it introduces other type mismatches.

**Comment by Milerius** [comment]
> @CryptoAssassin1 According to [Trust Doc](https://developer.trustwallet.com/developer/develop-for-trust/browser-extension), Trust Wallet Browser Extension currently supports only Ethereum & EVM chains, and support for Cosmos is still in progress.

Doc is just outdated it's not true

**Comment by isreallee82** [comment]
will look into that @CryptoAssassin1

**Comment by MaxMillington** [comment]
Hi @isreallee82 any update on this?

**Comment by isreallee82** [comment]
sorry for the late response to this fix @MaxMillington I have not been able to test my changes as trust wallet get stuck loading on my browser, I will try again today and give feedback

**Comment by MaxMillington** [comment]
Thanks @isreallee82 . Let us know what sorts of issues you run into and maybe we can help.

**Comment by isreallee82** [comment]
Screenshot 2024-03-13 at 2 25 13 PM

my chrome browser is stuck on this trust wallet loading

**Comment by isreallee82** [comment]
Screenshot 2024-03-13 at 2 25 13 PM

**Comment by isreallee82** [comment]
okay seems to work now

**Comment by CryptoAssassin1** [comment]
> will look into that @CryptoAssassin1

@isreallee82 the issue is still there.

**Comment by kamal-sutra** [comment]
@isreallee82 Please check the assets page, it is not released yet, might be on different branch.

image

**Comment by CryptoAssassin1** [comment]
@jonator can you please disable the assets page v2.

**Comment by CryptoAssassin1** [comment]
@jonator can you please disable the assets page v2.

**Comment by isreallee82** [comment]
@CryptoAssassin1 i am still reviewing why that is not working since algoric-3 is included, it was supposed to but i am on it.

**Comment by isreallee82** [comment]
I am not sure I made any changes related to asses @kamal-sutra, I work with stage branch.

**Comment by isreallee82** [comment]
@MaxMillington @CryptoAssassin1 BLD and ITS are not on the trust extension but on mobile yet I figured thats why you get that error.

**Comment by isreallee82** [comment]
should I remove chainids not supported by the extension @MaxMillington

**Comment by MaxMillington** [comment]
@isreallee82 So are BLD and ITS removed? Should we re-review/re-test?

**Comment by isreallee82** [comment]
yes I removed algorics-3 since its not available on the extension supported chainIds @MaxMillington

**Comment by MaxMillington** [comment]
@isreallee82 Can you rebase with stage and push again? There is a build error and it is fixed on the stage branch

**Comment by isreallee82** [comment]
done @MaxMillington

**Comment by isreallee82** [comment]
is there a problem with the build @MaxMillington

**Review by kamal-sutra — APPROVED** [review]
All txs working fine. Good to merge.

**Review by DavideSegullo — COMMENTED** [review]
I just wanna point out that right now account switch doesn't works, I opened a thread here:
https://discord.com/channels/1098013147891249152/1156015685013602354/1214915212159881227

But I don't know when they'll add it

**Comment by CryptoAssassin1** [comment]
@isreallee82 70% to 80% tpken deposit/withdraw buttons are disabled can you please check.

**Comment by isreallee82** [comment]
a min @MaxMillington will check that Now

**Comment by CryptoAssassin1** [comment]
Thank you

**Comment by isreallee82** [comment]
@CryptoAssassin1 thats because they are not compatible with the extension

**Comment by isreallee82** [comment]
I will include some chainIds back

**Comment by isreallee82** [comment]
will push in a bit

**Comment by kamal-sutra** [comment]
@isreallee82 @DavideSegullo it shows keplr twice
image

**Comment by isreallee82** [comment]
Will remove duplicate @kamal-sutra

**Comment by isreallee82** [comment]
once deployed please reconfirm @kamal-sutra. Things should work ritght now.

**Comment by CryptoAssassin1** [comment]
@isreallee82 Trust mobile option is available on mobile and its connecting with trust wallet everything is loading when try to swap or do any other tx getting 404 error
Screenshot from 2024-03-18 23-40-38

**Comment by isreallee82** [comment]
Will look into that shortly @CryptoAssassin1

**Comment by isreallee82** [comment]
Please can you show more clearer screenshots of the error @CryptoAssassin1. Things shows normal on my end using a chrome on mobile

**Comment by isreallee82** [comment]
Please can you show more clearer screenshots of the error @CryptoAssassin1. Things shows normal on my end using a chrome on mobile

**Comment by CryptoAssassin1** [comment]
> Please can you show more clearer screenshots of the error @CryptoAssassin1. Things shows normal on my end using a chrome on mobile



When i try to sign tx using trust wallet I'm getting 404 error

**Comment by kamal-sutra** [comment]
> Will remove duplicate @kamal-sutra

duplicates are removed. looks fine now.

**Comment by isreallee82** [comment]
can you try now @CryptoAssassin1

**Comment by CryptoAssassin1** [comment]
Still getting Not found

**Comment by isreallee82** [comment]
I did few checks the best solution is to update the package @cosmos-kit/core

**Comment by isreallee82** [comment]
the current version 2.7.2 latest version 2.8.9 There has been different changes to the types over time

**Comment by isreallee82** [comment]
can you try now @CryptoAssassin1 @MaxMillington

**Comment by CryptoAssassin1** [comment]
@isreallee82 still same error

**Comment by isreallee82** [comment]
You mean not found @CryptoAssassin1 ?

**Comment by CryptoAssassin1** [comment]
Yes.

**Comment by isreallee82** [comment]
What link can I use to access @CryptoAssassin1

**Comment by CryptoAssassin1** [comment]
Download the Trust wallet mobile app.

**Comment by isreallee82** [comment]
I mean for the deployment for this pr

**Comment by isreallee82** [comment]
I will make a push real quick @CryptoAssassin1

**Comment by MaxMillington** [comment]
@isreallee82 fyi I was OOO last week. @CryptoAssassin1 I'm not sure if anyone approved deploys while I was out? Is it possible you were testing on an old preview deploy?

**Comment by isreallee82** [comment]
okay I made few adjustment @MaxMillington @CryptoAssassin1. Separated core- walletconnect for trust specifically.

**Comment by isreallee82** [comment]
its alright @MaxMillington

**Comment by CryptoAssassin1** [comment]
@MaxMillington maybe but i just test the April 1 deploy and still getting same error.
@MaxMillington can you please approve the deploy again.

**Comment by CryptoAssassin1** [comment]
@isreallee82 same not found error.
@kamal-sutra can you please check on your end.

**Comment by kamal-sutra** [comment]
@isreallee82 still facing connection issues and swap, deposit issues
https://github.com/osmosis-labs/osmosis-frontend/assets/116148606/85fef375-0508-41e9-b7ba-0105a3cc9468

**Comment by isreallee82** [comment]
I will dig more into the problem @CryptoAssassin1

**Comment by CryptoAssassin1** [comment]
Thank you

**Comment by MaxMillington** [comment]
Screenshot 2024-04-04 at 7 53 21 AM

@isreallee82 @CryptoAssassin1 FYI Trust Wallet team had this to say ^

**Comment by MaxMillington** [comment]
This error is just on mobile right? @CryptoAssassin1

**Comment by CryptoAssassin1** [comment]
Yes @MaxMillington

**Review by CryptoAssassin1 — APPROVED** [review]
BE is working fine @MaxMillington

**Comment by MaxMillington** [comment]
@isreallee82 perhaps we just need to hide the integration on mobile then?

**Comment by isreallee82** [comment]
Sure I will work on that @MaxMillington

**Comment by isreallee82** [comment]
I disable the mobile trust @MaxMillington. if later on cosmos manage to update their trust mobile integration, you can let me know, I will create a separate pr to modify.

**Comment by MaxMillington** [comment]
@CryptoAssassin1 can you test to make sure it's disabled on mobile? And give it a once over again? If all is good then we can merge

**Comment by CryptoAssassin1** [comment]
@MaxMillington its still building.
This branch had an error being deployed

**Comment by CryptoAssassin1** [comment]
@isreallee82 Trust mobile is still visible.
Screenshot from 2024-04-05 01-24-12
Screenshot from 2024-04-05 01-24-34

**Comment by MaxMillington** [comment]
@CryptoAssassin1 did you test it in an actual mobile device or just on desktop? Because trust wallet has a `mobileDisabled` flag that might check for more than just screen width

**Comment by CryptoAssassin1** [comment]
@MaxMillington yes i tested on Mobile Trust Mobile is still showing but when you connect to Trust Wallet Mobile it display the message wallet is not supported on mobile use desktop extension.

**Comment by isreallee82** [comment]
Would you like it not show atall @MaxMillington  @CryptoAssassin1  if that’s the case I might need to comment out trust-mobile related

**Comment by MaxMillington** [comment]
@isreallee82 Yea I think we should just not show it at all on mobile

**Comment by isreallee82** [comment]
alright will push that in a min

**Comment by vcoolish** [comment]
trust wallet mobile supports cosmos dapps both injected and walletconnect connections, do you have some logs we can look into?

**Comment by isreallee82** [comment]
I will like to give another shot at it @MaxMillington @CryptoAssassin1 
will push in a min

**Comment by isreallee82** [comment]
> I will like to give another shot at it @MaxMillington @CryptoAssassin1 will push in a min

Not found while trying to sign transaction and deposit

**Comment by MaxMillington** [comment]
> > I will like to give another shot at it @MaxMillington @CryptoAssassin1 will push in a min
> 
> Not found while trying to sign transaction and deposit

@isreallee82 Just so I understand, you are still getting a mobile error? If so, can you provide the logs so that @vcoolish can take a look?

**Comment by isreallee82** [comment]
Hmm it’s not building on verve @MaxMillington

**Comment by isreallee82** [comment]
please can you redeploy I just updated branch @MaxMillington

**Comment by MaxMillington** [comment]
@isreallee82 approved. give it 10 minutes

**Comment by MaxMillington** [comment]
Screenshot 2024-04-09 at 12 49 31 PM
@isreallee82 deploy failed ^^

**Comment by isreallee82** [comment]
oh I see @MaxMillington another folder is using a different version, I will update the version and test if its compatible

**Comment by isreallee82** [comment]
fixed @MaxMillington

**Comment by isreallee82** [comment]
Waiting for deployment @MaxMillington

**Comment by MaxMillington** [comment]
@isreallee82 Apologies for the slow reply. Deploying now

**Comment by isreallee82** [comment]
please @CryptoAssassin1 @kamal-sutra

**Comment by isreallee82** [comment]
> Screenshot 2024-04-22 at 11 22 38 AM
> @isreallee82 looks like a type error

this must be due to previous deployment on the master branch @MaxMillington

**Comment by kamal-sutra** [comment]
@isreallee82 it is still happening 
https://github.com/osmosis-labs/osmosis-frontend/assets/116148606/fa79ec0c-7f5c-4227-ab05-2fd8f2064dbf

**Comment by CryptoAssassin1** [comment]
@isreallee82 still getting not found error.

**Comment by isreallee82** [comment]
@MaxMillington I am not sure I am authorised to see the logs

**Comment by isreallee82** [comment]
In that case i think 

> Screenshot 2024-04-04 at 7 53 21 AM
> @isreallee82 @CryptoAssassin1 FYI Trust Wallet team had this to say ^

I think this should be true because I followed the same process with latest wallet core, trust wallet might support caller connect integration for mobile but not on cosmos yet

**Comment by CryptoAssassin1** [comment]
> trust wallet mobile supports cosmos dapps both injected and walletconnect connections, do you have some logs we can look into?

@isreallee82

**Comment by isreallee82** [comment]
> > trust wallet mobile supports cosmos dapps both injected and walletconnect connections, do you have some logs we can look into?
> 
> @isreallee82

@CryptoAssassin1 as I said earlier any logs will be within the verve deployment which I can’t access

**Comment by MaxMillington** [comment]
@isreallee82 are you not seeing the same issue locally?

**Comment by isreallee82** [comment]
> @isreallee82 are you not seeing the same issue locally?
No mobile wallets are not coming up locally @MaxMillington

**Comment by isreallee82** [comment]
You are authorized to see the vercel logs for this pr right @CryptoAssassin1, if so you could view the logs real time and try reproducing the error so we can know what the actual error is.
What did you think?

**Comment by CryptoAssassin1** [comment]
You can see all relevant logs from running the code locally on your machine @isreallee82

**Comment by isreallee82** [comment]
yea I could test that with extension version but since we working with mobile I am not sure localhost will work since I need to connect with a url but let me try figure other options @CryptoAssassin1

**Comment by isreallee82** [comment]
figured I can run with vercel I will try that and give a feedback

**Comment by isreallee82** [comment]
For some reason asset page don’t appear on my deployment https://osmosis-frontend-eauxtnpua-ajayi-israels-projects-d0221277.vercel.app/?to=OSMO&from=ATOM is there an additional configuration @MaxMillington

**Comment by MaxMillington** [comment]
@isreallee82 shouldn't need any additional configuration. I'm not able to see that deploy though, it says I don't have permission.

**Comment by isreallee82** [comment]
oh okay 
yea thats because its deployed on my Vercel account actually @MaxMillington

**Comment by MaxMillington** [comment]
@CryptoAssassin1 @kamal-sutra can you test once build is finished?

**Comment by isreallee82** [comment]
As previously mentioned to @CryptoAssassin1 and @MaxMillington, assets do not appear in the local build or even when deployed on a VPS; they only show up on your Vercel deployment. The most effective approach for review is attempting to reproduce the error while examining the logs on the Vercel deployment. In my experience, 'Note found' may indicate an outdated URL endpoint for transactions on Cosmos.

**Comment by CryptoAssassin1** [comment]
@isreallee82 still same error on the old build.
@MaxMillington please approve the new build.

**Comment by CryptoAssassin1** [comment]
@isreallee82 still same error.

**Comment by isreallee82** [comment]
It’s weird it’s deploying here but I have not been able to build on my local machine for like a week now even on stage branch

**Comment by isreallee82** [comment]
I will try rerun with todays commiton stage branch @CryptoAssassin1

**Comment by isreallee82** [comment]
Screenshot 2024-05-15 at 10 09 27 PM

@MaxMillington @CryptoAssassin1

**Comment by isreallee82** [comment]
Screenshot 2024-05-15 at 10 11 21 PM

**Comment by isreallee82** [comment]
I was able to reproduce the error locally but no log @MaxMillington @CryptoAssassin1

**Comment by isreallee82** [comment]
here is the request its trying to make will look into it and give feedback @MaxMillington  @CryptoAssassin1 http://192.168.8.101:3000/assets/trust/WC?requestld=1716132142102126&sessionTopic=2f50a7657bb43b-b6a715cbe5c0b15b32ffa743a6ecb19f0c4275516bfa73c870

**Comment by isreallee82** [comment]
@vcoolish can you reconfirm http://192.168.8.101:3000/assets/trust/WC?requestld=1716132142102126&sessionTopic=2f50a7657bb43b-b6a715cbe5c0b15b32ffa743a6ecb19f0c4275516bfa73c870.  cc @MaxMillington @CryptoAssassin1

**Comment by isreallee82** [comment]
I looking into the code and did more digging aside the fact the links in trust-mobile were outdated,  the trust registry doesnt seem to support the signing method for trust-mobile.

**Comment by isreallee82** [comment]
hi I still await a response regarding my last comment cc @MaxMillington cc @CryptoAssassin1  cc @vcoolish also. I keep merging so the pr is in sync with the current stage branch @MaxMillington

**Comment by isreallee82** [comment]
Alright, great. @MaxMillington and I will wait on their response. I think it's a good call for now, and I'll also wait on your decision to disable mobile pending the update from Trust Wallet.

**Comment by isreallee82** [comment]
since I made the initial progress @MaxMillington and already integrated extension part of trust-wallet, do you feel I should proceed with pushing commit for disabling trust mobile?

**Comment by MaxMillington** [comment]
@isreallee82 yes, go ahead and disable for mobile

**Comment by isreallee82** [comment]
Related code for trust mobile have been commented @MaxMillington

**Comment by isreallee82** [comment]
can you approve @MaxMillington

**Comment by isreallee82** [comment]
Can you recheck @MaxMillington since changes have been made overtime cc @CryptoAssassin1 cc @kamal-sutra

**Comment by MaxMillington** [comment]
@CryptoAssassin1 and @kamal-sutra to test

**Comment by kamal-sutra** [comment]
@isreallee82 Please check the blinking issue in the video
[Screencast from 11-06-24 08:34:39 PM IST.webm](https://github.com/osmosis-labs/osmosis-frontend/assets/116148606/b9b09a03-80ae-419c-b8ff-ec4dc7699d7d)

**Comment by CryptoAssassin1** [comment]
I can reproduce the blinking issue.

**Comment by isreallee82** [comment]
just updated the branch with latest stage, must have been the last merge can you reapprove @MaxMillington

**Comment by isreallee82** [comment]
I updated the trust extension and removed the changes made to yarn.lock @MaxMillington

**Comment by isreallee82** [comment]
I noticed the flicking still persist my guess must be wallet connect core package not compatible with some other package, since I used a newer version, but it’s okay since the other pr provides a better solution @MaxMillington its been nice working on this cc @CryptoAssassin1  cc @kamal-sutra.

**Comment by MaxMillington** [comment]
Thanks @isreallee82. We will be posting more bounties in the near future. Feel free to give those a try as well.

Closing as this issue was fixed in: https://github.com/osmosis-labs/osmosis-frontend/pull/3231

**Pull request closed without merging**
